### PR TITLE
Test-end-2-end-3rd-phase

### PR DESCRIPTION
**Pull request by diegolagospagopa**
### List of changes

added k8s and postgres info + added keyvault properties

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [x] DEV
- [x] UAT
- [ ] PROD

### CODE REVIEW

File: src/k8s/99_main.tf [hunk 1]
```diff
@@ -22,8 +22,8 @@ terraform {
 }
 
 data "azurerm_kubernetes_cluster" "aks_cluster" {
-  name                = format("%s-aks", local.project)
-  resource_group_name = format("%s-aks-rg", local.project)
+  name                = "usrreg-${var.env_short}-aks"
+  resource_group_name = "usrreg-${var.env_short}-aks-rg"
```
Comment (pasqualedevita) [inline src/k8s/99_main.tf]: why not local.project?

File: src/pillar/01_keyvault.tf [hunk 2]
```diff
@@ -12,3 +12,48 @@ data "azurerm_key_vault_secret" "monitor_notification_email" {
...
+
+#
+# Added new secrets
+#
+resource "random_password" "autogenerated_password_values" {
+  count = (
+    var.keyvault_autogenerated_secrets != null || var.keyvault_autogenerated_secrets != []
+    ) ? length(var.keyvault_autogenerated_secrets) : 0
+
+  length           = 12
```
Comment (pasqualedevita) [inline src/pillar/01_keyvault.tf]:
```suggestion
  length           = 24
```

File: src/pillar/01_keyvault.tf [hunk 3]
```diff
@@ -12,3 +12,48 @@ data "azurerm_key_vault_secret" "monitor_notification_email" {
...
   key_vault_id = data.azurerm_key_vault.kv.id
 }
+
+#
+# Added new secrets
+#
+resource "random_password" "autogenerated_password_values" {
+  count = (
+    var.keyvault_autogenerated_secrets != null || var.keyvault_autogenerated_secrets != []
+    ) ? length(var.keyvault_autogenerated_secrets) : 0
```
Comment (pasqualedevita) [inline src/pillar/01_keyvault.tf]: probably more simple
```
resource "random_password" "autogenerated_password_values" {
  count = length(var.keyvault_autogenerated_secrets)
  ...
}

```
and in variables.tf file set default value to
```
variable "keyvault_autogenerated_secrets" {
  type = list(string)
  default = []
}
```

File: src/pillar/01_keyvault.tf [hunk 4]
```diff
@@ -12,3 +12,48 @@ data "azurerm_key_vault_secret" "monitor_notification_email" {
...
+
+  length           = 12
+  special          = true
+  override_special = "!$#%"
+}
+
+resource "azurerm_key_vault_secret" "autogenerated_secrets" {
```
Comment (pasqualedevita) [inline src/pillar/01_keyvault.tf:27]: same

File: src/pillar/01_keyvault.tf [hunk 5]
```diff
@@ -12,3 +12,48 @@ data "azurerm_key_vault_secret" "monitor_notification_email" {
...
+  count = (
+    var.keyvault_autogenerated_secrets != null || var.keyvault_autogenerated_secrets != []
+    ) ? length(var.keyvault_autogenerated_secrets) : 0
+
+  name         = var.keyvault_autogenerated_secrets[count.index]
+  value        = random_password.autogenerated_password_values[count.index].result
+  key_vault_id = data.azurerm_key_vault.kv.id
+
+  expiration_date = "2022-12-31T00:00:00Z"
```
Comment (pasqualedevita) [inline src/pillar/01_keyvault.tf]: we can remove expiration_date

File: src/pillar/01_keyvault.tf (hunk 2 above)
Comment (diegolagospagopa) [reply src/pillar/01_keyvault.tf]: Done

File: src/pillar/01_keyvault.tf (hunk 4 above)
Comment (diegolagospagopa) [reply src/pillar/01_keyvault.tf:27]: Done

File: src/pillar/01_keyvault.tf (hunk 5 above)
Comment (diegolagospagopa) [reply src/pillar/01_keyvault.tf]: Done

File: src/pillar/01_keyvault.tf (hunk 3 above)
Comment (diegolagospagopa) [reply src/pillar/01_keyvault.tf]: done

File: src/k8s/99_main.tf (hunk 1 above)
Comment (diegolagospagopa) [reply src/k8s/99_main.tf]: done

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**